### PR TITLE
Enhancement: auto calculate padding for overflowing grid labels

### PR DESCRIPTION
**Pull request by Mgetz10**
## No Ticket

In an effort to never overlap data with grid when "Only show top prefix/suffix" is activated - auto generate padding if gridline is above maxValue

## Testing Steps
1. Create of view linear chart vertical chart with "Only show top prefix/suffix" activated
2. Observe the label does not overlap data

## Self Review

- I have added testing steps for reviewers
- I have commented my code, particularly in hard-to-understand areas
- My changes generate no new warnings
- New and existing unit tests are passing

### Before
Screenshot 2024-11-26 at 2 40 04 PM

### After
Screenshot 2024-11-26 at 2 40 11 PM

### CODE REVIEW

File: packages/chart/src/components/LinearChart.tsx [hunk 1]
```diff
@@ -100,6 +100,7 @@ const LinearChart = forwardRef<SVGAElement, LinearChartProps>(({ parentHeight, p
   const [animatedChart, setAnimatedChart] = useState(false)
   const [point, setPoint] = useState({ x: 0, y: 0 })
   const [suffixWidth, setSuffixWidth] = useState(0)
+  const [autoPadding, setAutoPadding] = useState(0)
```
Comment (adamdoe) [inline packages/chart/src/components/LinearChart.tsx:103]: I hate to make this name longer, but I wonder if we should explicitly state what type of autoPadding this is (ie. yAxisAutoPadding) should we introduce other automatic padding into the mix.

File: packages/chart/src/components/LinearChart.tsx [hunk 2]
```diff
@@ -114,12 +115,73 @@ const LinearChart = forwardRef<SVGAElement, LinearChartProps>(({ parentHeight, p
     freezeOnceVisible: false
   })
 
+  // Pre render functions
+  const countNumOfTicks = (axis, max) => {
```
Comment (adamdoe) [inline packages/chart/src/components/LinearChart.tsx:119]: nit: Can we move this into `/helpers/countNumOfTicks.ts`?

File: packages/chart/src/components/LinearChart.tsx [hunk 3]
```diff
@@ -114,12 +115,73 @@ const LinearChart = forwardRef<SVGAElement, LinearChartProps>(({ parentHeight, p
...
+  }
+
   // VARS/MEMOS
   const shouldAbbreviate = true
   const isHorizontal = orientation === 'horizontal' || config.visualizationType === 'Forest Plot'
   const isLogarithmicAxis = config.yAxis.type === 'logarithmic'
   const isForestPlot = visualizationType === 'Forest Plot'
   const suffixHasNoSpace = !suffix.includes(' ')
+  const padding = orientation === 'horizontal' ? Number(config.xAxis.size) : Number(config.yAxis.size)
+  const fontSize = { small: 16, medium: 18, large: 20 }
```
Comment (adamdoe) [inline packages/chart/src/components/LinearChart.tsx:184]: Can we remove this and use `packages/core/helpers/cove/fontSettings.ts`. I think this is one of those repeated values we need to remove and reducing the amount of repetition or removing it altogether in favor of CSS (if possible) would be helpful.

**Review by adamdoe — CHANGES_REQUESTED** [review]

File: packages/chart/src/components/LinearChart.tsx (hunk 3 above)
Comment (Mgetz10) [reply packages/chart/src/components/LinearChart.tsx:184]: @adamdoe is ok if we keep this here and save the refactor for a more relevant change? Its not really related to work i did - i just moved it up from a lower part of the component in an attempt to keep the variables together

Comment (adamdoe) [reply packages/chart/src/components/LinearChart.tsx:184]: Sure, thats fine!

Comment (Mgetz10) [reply packages/chart/src/components/LinearChart.tsx:184]: @adamdoe Oh sorry I didn't realize how easy the request was! adding in the change 🫡

**Comment by Mgetz10** [comment]
@adamdoe @jayb Ready for review

**Comment by Mgetz10** [comment]
@adamdoe NVM! new bug found

**Review by jayb — APPROVED** [review]
Works great!

**Comment by Mgetz10** [comment]
@adamdoe ok ready for you when you have a moment

**Review by adamdoe — APPROVED** [review]

**Pull request merged**